integr8ly/integreatly-operator
# INTLY-4489 - Upgrade Fuse online and fix service discovery

**JIRAs:**
https://issues.redhat.com/browse/INTLY-4489
https://issues.redhat.com/browse/INTLY-4527 (motivation for the Fuse upgrade)

**What:**
- upgraded Fuse to 7.5, manifests are based on [internal repo](http://pkgs.devel.redhat.com/cgit/containers/fuse-online-operator/tree/manifests/7.5.0?h=fuse-7.5-openshift-rhel-7)
- fixed service discovery of (shared) Fuse OpenAPI integrations with 3Scale
- dependency update (separate commit for easy filtering out)
- small improvement to tests code (also separate commit)


**Verification steps:**
Install Integreatly with this PR. 3Scale and Fuse installation needs to complete to proceed with verification.
Verify that new version(7.5) of fuse is displayed in Installation CR status.
Create a htpasswd IDP (`./scripts/setup-htpass-idp.sh`)
Login into OpenShift as customer-admin while the operator is running (so the users are synced into 3Scale)
Open Fuse Online console and create an OpenAPI based integration:
![Screenshot 2020-01-08 at 18 08 51](https://user-images.githubusercontent.com/1605799/72066122-0cfa1300-32e0-11ea-9db0-5ae9f54f5ea0.png)
Value for the URL field: `https://raw.githubusercontent.com/syndesisio/syndesis-quickstarts/master/api-provider/task-api.json`
Click Next (twice?) and then click "Publish" (on the top right).
Wait until the integration is built and deployed successfully.
Login into 3Scale console (`https://3scale-admin...`)
Click on "New Product":
![Screenshot 2020-01-08 at 17 19 56](https://user-images.githubusercontent.com/1605799/72066292-75e18b00-32e0-11ea-840d-565637e51fb5.png)
Select "Import from OpenShift" (there might be an "Authenticate" button next to this option, click on that first and approve the permissions)
Namespace and Name fields should become populated with a discovered services:
![Screenshot 2020-01-08 at 18 19 42](https://user-images.githubusercontent.com/1605799/72066609-56972d80-32e1-11ea-999b-1d11477f3cf8.png)
Click "Create Product" and verify there are no errors displayed.
